dexcore/core.go

```diff
@@ -0,0 +1,92 @@
+package dexcore
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"decred.org/dcrdex/client/core"
+	"decred.org/dcrdex/dex"
+	libutils "github.com/crypto-power/cryptopower/libwallet/utils"
+)
+
+type Core interface {
+	Run(ctx context.Context)
+	Ready() <-chan struct{}
+	IsInitialized() bool
+	InitializeClient(pw, seed []byte) error
+	DiscoverAccount(dexAddr string, pass []byte, certI any) (*core.Exchange, bool, error)
+	BondsFeeBuffer(assetID uint32) uint64
+	PostBond(form *core.PostBondForm) (*core.PostBondResult, error)
+	NotificationFeed() <-chan core.Notification
+	Login(pw []byte) error
+}
+
+// New returns a *DEXCore that satisfies Core.
+func New(root, lang string, net libutils.NetworkType) (*DEXCore, error) {
+	dexNet := toDEXNet(string(net))
+	network, err := dex.NetFromString(dexNet)
+	if err != nil {
+		return nil, fmt.Errorf("dex.NetFromString error: %w", err)
+	}
+
+	dbPath := filepath.Join(root, string(net), "dexc.db")
```

Comment:
I have a guess why you're using string(net) instead of dexNet, but add a comment.